### PR TITLE
Deduplicate `ActionMenu.tsx`

### PR DESCRIPTION
**Pull request by iansan5653**
While looking into https://github.com/primer/react/issues/3712, I noticed that `ActionMenu.tsx` is duplicated. It exists once in `src/` and once in `src/ActionMenu`. The `src/ActionMenu.tsx` file is the one being exported because it takes priority over `src/ActionMenu/index.ts`, but the `src/ActionMenu/ActionMenu.tsx` file is the one being tested and rendered in Storybook. This strikes me as particularly risky as the two could diverge over time (they already have in that the `aria-labelledby` prop was only added to the file in `src/`). 

This appears to have been a leftover from some rearranging of files.

This PR cleans up the duplication by merging the two files and deleting the one in `src/`.

- Closes https://github.com/primer/react/issues/2992

### CODE REVIEW

File: src/ActionMenu/ActionMenu.tsx [hunk 1]
```diff
@@ -89,7 +89,12 @@ type MenuOverlayProps = Partial<OverlayProps> &
      */
     children: React.ReactNode
   }
-const Overlay: React.FC<React.PropsWithChildren<MenuOverlayProps>> = ({children, align = 'start', ...overlayProps}) => {
+const Overlay: React.FC<React.PropsWithChildren<MenuOverlayProps>> = ({
+  children,
+  align = 'start',
+  'aria-labelledby': ariaLabelledby,
```
Comment (iansan5653) [inline src/ActionMenu/ActionMenu.tsx:95]: This was the only change that needed to be pulled in from the deleted file. (added in https://github.com/primer/react/pull/1759)

**Review by broccolinisoup — COMMENTED** [review]
Hi @iansan5653 👋🏻 Thanks so much for pushing this PR! I found that we even had an [issue](https://github.com/primer/react/issues/2992) created to delete some duplicate files that includes `ActionMenu.tsx` - I updated the description of the issue as we revereted some Dialog changes so only file that is duplicated for now is `ActionMenu.tsx`

Your changes is looking great, my only concern is that there is a [manual import at dotcom from a file path](https://github.com/github/github/blob/c0cf905416f2f84c81a6f5e92e37b64d9434f5e7/ui/packages/memex/src/client/components/tracked-by/tracked-by-missing-issues-button.tsx#L5C26-L5C49). If we could fix that at dotcom before introducing this changes in primer/react side, I think it should be good. Let me know if I am misinterpreting anything! 

Feel free to update your description with`Closes https://github.com/primer/react/issues/2992` as well.

Thanks again for pushing this PR 🙌🏻

**Comment by iansan5653** [comment]
> my only concern is that there is a manual import at dotcom from a file path

I think this is the import in question:

```tsx
import {ActionMenu} from '@primer/react/lib-esm/ActionMenu'
```

This should still work with this change, because it will automatically resolve to `ActionMenu/index.tsx` instead of `ActionMenu.tsx` now that the file is deleted. This is also why we don't have to change any import paths within `primer/react` either.

**Comment by iansan5653** [comment]
The diff is changing as I work! The new one highlights a small difference in `ActionMenuButtonProps`; easy to resolve.

```diff
diff --git a/src/ActionMenu.tsx b/src/ActionMenu/ActionMenu.tsx
index 2f55d433..b84b1a65 100644
--- a/src/ActionMenu.tsx
+++ b/src/ActionMenu/ActionMenu.tsx
@@ -1,14 +1,14 @@
 import React from 'react'
 import {TriangleDownIcon} from '@primer/octicons-react'
-import {AnchoredOverlay, AnchoredOverlayProps} from './AnchoredOverlay'
-import {OverlayProps} from './Overlay'
-import {useProvidedRefOrCreate, useProvidedStateOrCreate, useMenuKeyboardNavigation} from './hooks'
-import {Divider} from './ActionList/Divider'
-import {ActionListContainerContext} from './ActionList/ActionListContainerContext'
-import {Button, ButtonProps} from './Button'
-import {useId} from './hooks/useId'
-import {MandateProps} from './utils/types'
-import {ForwardRefComponent as PolymorphicForwardRefComponent} from './utils/polymorphic'
+import {AnchoredOverlay, AnchoredOverlayProps} from '../AnchoredOverlay'
+import {OverlayProps} from '../Overlay'
+import {useProvidedRefOrCreate, useProvidedStateOrCreate, useMenuKeyboardNavigation} from '../hooks'
+import {Divider} from '../ActionList/Divider'
+import {ActionListContainerContext} from '../ActionList/ActionListContainerContext'
+import {Button, ButtonProps} from '../Button'
+import {useId} from '../hooks/useId'
+import {MandateProps} from '../utils/types'
+import {ForwardRefComponent as PolymorphicForwardRefComponent} from '../utils/polymorphic'
 
 export type MenuContextProps = Pick<
   AnchoredOverlayProps,
@@ -73,7 +73,7 @@ const Anchor = React.forwardRef<HTMLElement, ActionMenuAnchorProps>(({children,
 })
 
 /** this component is syntactical sugar 🍭 */
-export type ActionMenuButtonProps = ButtonProps & {
+export type ActionMenuButtonProps = Omit<ButtonProps, 'children'> & {
   children: React.ReactNode
 }
 const MenuButton = React.forwardRef(({...props}, anchorRef) => {
@@ -91,12 +91,7 @@ type MenuOverlayProps = Partial<OverlayProps> &
      */
     children: React.ReactNode
   }
-const Overlay: React.FC<React.PropsWithChildren<MenuOverlayProps>> = ({
-  children,
-  align = 'start',
-  'aria-labelledby': ariaLabelledby,
-  ...overlayProps
-}) => {
+const Overlay: React.FC<React.PropsWithChildren<MenuOverlayProps>> = ({children, align = 'start', ...overlayProps}) => {
   // we typecast anchorRef as required instead of optional
   // because we know that we're setting it in context in Menu
   const {anchorRef, renderAnchor, anchorId, open, onOpen, onClose} = React.useContext(MenuContext) as MandateProps<
@@ -124,7 +119,7 @@ const Overlay: React.FC<React.PropsWithChildren<MenuOverlayProps>> = ({
           value={{
             container: 'ActionMenu',
             listRole: 'menu',
-            listLabelledBy: ariaLabelledby || anchorId,
+            listLabelledBy: anchorId,
             selectionAttribute: 'aria-checked', // Should this be here?
             afterSelect: onClose,
           }}
```

**Review by joshblack — APPROVED** [review]

**Comment by broccolinisoup** [comment]
👋🏻 @iansan5653 integration tests are [complaining](https://github.com/github/github/actions/runs/6145416379/job/16672795503?pr=288076) about not being able to resolve `@primer/react/lib-esm/ActionMenu` - should we correct the path at dotcom?

**Comment by siddharthkp** [comment]
I'd vote for quickly changing the imports in dotcom instead of reverting the PR :)

**Comment by siddharthkp** [comment]
> I think we should time box option 3 and use option 1 as a backup. I can work on this

I have a PR ready for review :) https://github.com/primer/react/pull/3740

**Pull request merged**